athena-docdb/src/main/java/com/amazonaws/athena/connectors/docdb/QueryUtils.java

```
@@ -248,6 +268,343 @@ public static Document parseFilter(String filter)
+    {
+        if (expression == null) {
+            return new Document();
+        }
+
+        // Handle leaf nodes (individual predicates like job_title = 'Engineer')
+        if (expression.isLeaf()) {
+            // Convert leaf predicate to MongoDB document using existing convertColumnPredicatesToDoc logic
+            // This ensures all existing optimizations (like $in for multiple EQUAL values) are preserved
+            ColumnPredicate predicate = expression.getLeafPredicate();
+            return convertColumnPredicatesToDoc(predicate.getColumn(),
+                    Collections.singletonList(predicate));
+        }
+
+        // Handle logical operators (AND/OR nodes with children)
+        // Recursively convert each child expression to MongoDB document
+        List<Document> childDocuments = new ArrayList<>();
+        for (LogicalExpression child : expression.getChildren()) {
+            Document childDoc = makeQueryFromLogicalExpression(child);
+            if (childDoc != null && !childDoc.isEmpty()) {
+                childDocuments.add(childDoc);
+            }
+        }
+
+        if (childDocuments.isEmpty()) {
+            return new Document();
+        }
+        if (childDocuments.size() == 1) {
+            // Single child - no need for logical operator wrapper
+            return childDocuments.get(0);
+        }
+
+        // Apply the logical operator to combine child documents
+        // Example: AND → {"$and": [child1, child2]}, OR → {"$or": [child1, child2]}
+        return switch (expression.getOperator()) {
+            case AND -> new Document(AND_OP, childDocuments); // {"$and": [{"col1": "val1"}, {"col2": "val2"}]}
+            case OR -> new Document(OR_OP, childDocuments);   // {"$or": [{"col1": "val1"}, {"col2": "val2"}]}
+            default ->
+                // For unknown operators, default to OR for safety
+                    new Document(OR_OP, childDocuments);
```

should we throw exception when an unknown logic operator is used ? query results from using OR operator might not be correct.

---

we cannot fallback to a default here. If we ever get to a point where an operator is pushed down to this layer that is not applicable to the source, then we messed up in communicating the capabilities to the engine.

---

Throwing UnsupportedOperationException now.